src/Core/ServerSettings.h

```diff
@@ -97,6 +97,8 @@ namespace DB
     M(UInt64, dns_cache_max_entries, 10000, "Internal DNS cache max entries.", 0) \
     M(Int32,  dns_cache_update_period, 15, "Internal DNS cache update period in seconds.", 0) \
     M(UInt32, dns_max_consecutive_failures, 10, "Max DNS resolve failures of a hostname before dropping the hostname from ClickHouse DNS cache.", 0) \
+    M(Bool, dns_allow_resolve_names_to_ipv4, true, "Allows resolve names to ipv4 addresses.", 0) \
```

Comment:
How about keeping in a separate section `<dns_resolver>...</dns_resolver>` ? To have everything in one place.

Reply:
I agree, it's good to keep `dns_resolver` setting under the separate section. But as for me this section also should contain others dns options, because logically `dns_cache` options are subset of `dns_resolver` : https://github.com/ClickHouse/ClickHouse/blob/9f34daa98d538f0399a42100286bbe1d65218d02/src/Core/ServerSettings.h#L96-L99   

Suppose it should be done in separate pr